File: problems/Python/017_write_is_even/Test/is_even_tests.py
import importlib.util, sys

def load(fp):
    spec = importlib.util.spec_from_file_location("s", fp)
    m = importlib.util.module_from_spec(spec); spec.loader.exec_module(m); return m

def run_tests(fp):
    f = load(fp).is_even
    cases = [
        (4, True, "even positive"),
        (7, False, "odd positive"),
        (0, True, "zero is even"),
        (-2, True, "negative even"),
        (-3, False, "negative odd"),
        (100, True, "large even"),
        (1, False, "one is odd"),
    ]
    passed=failed=0; results=[]
    for n,exp,label in cases:
        try:
            r=f(n)
            if r==exp: passed+=1; results.append({"test":label,"status":"pass"})
            else: failed+=1; results.append({"test":label,"status":"fail","expected":exp,"got":r,"input":n})
        except Exception as e: failed+=1; results.append({"test":label,"status":"error","message":str(e),"input":n})
    return {"passed":passed,"failed":failed,"total":len(cases),"results":results}

if __name__=="__main__":
    print(run_tests(sys.argv[1] if len(sys.argv)>1 else "starter.py"))

File: problems/TypeScript/008_is_even/Test/is_even_tests.ts
import * as path from "path";
async function load(fp: string) { return await import(path.resolve(fp.replace(/\.ts$/,""))); }
async function runTests(fp: string) {
    const m = await load(fp); const f = m.isEven;
    const cases: [number,boolean,string][] = [[0,true,"zero"],[2,true,"two"],[4,true,"four"],[1,false,"one"],[7,false,"seven"],[100,true,"hundred"],[-2,true,"neg even"],[-3,false,"neg odd"]];
    let passed=0,failed=0; const results: any[]=[];
    for(const [n,exp,label] of cases){try{const r=f(n);if(r===exp){passed++;results.push({test:label,status:"pass"})}else{failed++;results.push({test:label,status:"fail",expected:exp,got:r})}}catch(e:any){failed++;results.push({test:label,status:"error",message:e.message})}}
    return {passed,failed,total:cases.length,results};
}
const fp=process.argv[2]??"starter.ts"; runTests(fp).then(o=>console.log(JSON.stringify(o,null,2)));
